knowledge-graph/src/main/scala/ch/datascience/knowledgegraph/graphql/lineage/LineageFinder.scala

```diff
@@ -91,97 +87,61 @@ class IOLineageFinder(
-         |  }
-         |  {
-         |    ?entity prov:qualifiedGeneration/prov:activity ?activity ;
-         |            rdfs:label ?target_label .
-         |    ?activity rdfs:comment ?source_label .
-         |    FILTER NOT EXISTS {?activity rdf:type wfprov:WorkflowRun}
-         |    FILTER EXISTS {?activity rdf:type wfprov:ProcessRun}
-         |    BIND (?entity AS ?target)
-         |    BIND (?activity AS ?source)
-         |  } UNION {
-         |    ?activity prov:qualifiedUsage/prov:entity ?entity ;
-         |              rdfs:comment ?target_label .
-         |    ?entity rdfs:label ?source_label .
-         |    FILTER NOT EXISTS {?activity rdf:type wfprov:WorkflowRun}
-         |    FILTER EXISTS {?activity rdf:type wfprov:ProcessRun}
-         |    BIND (?activity AS ?target)
-         |    BIND (?entity AS ?source)
-         |  }
-         |}""".stripMargin
-
-    private def withFilterOn(projectPath:   ProjectPath,
-                             maybeCommitId: Option[CommitId],
-                             maybeFilePath: Option[FilePath]): String =
-      s"""
-         |      ?qentity dcterms:isPartOf|schema:isPartOf ?project .
-         |      FILTER (?project = <${renkuBaseUrl / projectPath}>)
-         |      ${filterOn(maybeCommitId)}
-         |      ${filterOn(maybeCommitId, maybeFilePath)}
-         |""".stripMargin
-
-    private def filterOn(maybeCommitId: Option[CommitId]): String =
-      maybeCommitId
-        .map { commitId =>
-          s"""
-             |    ?qentity (prov:qualifiedGeneration/prov:activity | ^prov:entity/^prov:qualifiedUsage) ?qactivity .
-             |    FILTER (?qactivity = <file:///commit/$commitId>)
-       """.stripMargin
-        }
-        .getOrElse("")
-
-    private def filterOn(maybeCommitId: Option[CommitId], maybeFilePath: Option[FilePath]): String =
-      (maybeCommitId, maybeFilePath)
-        .mapN { (commitId, filePath) =>
-          s"""
-             |    FILTER (?qentity = <file:///blob/$commitId/$filePath>)
-       """.stripMargin
-        }
-        .getOrElse("")
-  }
+  private def query(projectPath: ProjectPath, commitId: CommitId, filePath: FilePath): String =
+    s"""
+       |PREFIX prov: <http://www.w3.org/ns/prov#>
+       |PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+       |PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+       |PREFIX wfdesc: <http://purl.org/wf4ever/wfdesc#>
+       |PREFIX wf: <http://www.w3.org/2005/01/wf/flow#>
+       |PREFIX wfprov: <http://purl.org/wf4ever/wfprov#>
+       |PREFIX foaf: <http://xmlns.com/foaf/0.1/>
+       |PREFIX schema: <http://schema.org/>
+       |PREFIX dcterms: <http://purl.org/dc/terms/>
+       |
+       |SELECT ?target ?source ?target_label ?source_label
+       |WHERE {
+       |  {
+       |    SELECT ?entity
+       |    WHERE {
+       |      ?qentity dcterms:isPartOf|schema:isPartOf ?project .
+       |      FILTER (?project = <${renkuBaseUrl / projectPath}>)
```

Comment:
I don't understand this - why not put the `<${renkuBaseUrl / projectPath}>` in the line above instead of using `FILTER`?

Reply:
Good catch. Lines collapsed 👍